File: Sorting/Java/src/lib/Sorting_Lib.java
public class Sorting_Lib {
  public static int[] insertionSort(int[] listToSort){
    int[] A = listToSort;
    int i = 1;
    while(i < A.length){
      int x = A[i];
      int j = i - 1;
      while((j >= 0) && (A[j] > x)){
        A[j+1] = A[j];
        j = j - 1;
      }
      A[j+1] = x;
      i = i + 1;
    }
    return A;
  }
}

File: Sorting/Javascript/src/lib/sorting_lib.js
export function insertionSort(listToSort){
  let A = listToSort;
  let i = 0;
  while(i < A.length){
    let x = A[i];
    let j = i - 1;
    while((j >= 0) && (A[j] > x)){
      A[j + 1] = A[j];
      j = j - 1;
    }
    A[j + 1] = x;
    i = i + 1;
  }
  return A;
}
